src/evict/evict_page.c

```diff
@@ -500,6 +518,22 @@ __evict_child_check(WT_SESSION_IMPL *session, WT_REF *parent)
                               */
             if (!__wt_atomic_casv8(&child->state, WT_REF_DELETED, WT_REF_LOCKED))
                 return (__wt_set_return(session, EBUSY));
+            /*
+             * Insert a read/read barrier so we're guaranteed the page_del state we read below comes
+             * after the locking operation on the ref state and therefore after the previous unlock
+             * of the ref. Otherwise we might read an inconsistent view of the page deletion info,
+             * and while many combinations are harmless and would just lead us to falsely refuse to
+             * evict, some (e.g. reading committed as true and a stale durable timestamp from before
+             * it was set by commit) are not.
+             *
+             * Note that while ordinarily a lock acquire should have an acquire (read/any) barrier
+             * after it, because we are only reading the write part is irrelevant and a read/read
+             * barrier is sufficient.
+             *
+             * FUTURE: this and the CAS should be rolled into a WT_REF_TRYLOCK macro.
```

Comment:
Can you create a ticket to track this work?

Reply:
Yeah, probably a good idea; it's WT-9780. Updated the comments in [9f4de66](https://github.com/wiredtiger/wiredtiger/pull/8188/commits/9f4de669cdc9003b518d3a587f95176468058569)